### PR TITLE
feat: question ingest script

### PR DESCRIPTION
**Pull request by dextertanyj**

### CODE REVIEW

**Review by junlong4321 — APPROVED** [review]
LGTM. Need to check regarding 3 things.

1. We might need to cast the test cases into a single string to render on the Submissions tab
2. Do I remove the questionFrontendId, titleSlug, similarQuestions, exampleTestCases
3. Do I commit the the Leetcode questions?

File: serverless/ingest.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+def usage(name: str):
+    logging.warn("Usage: %s <file> [<environment file>]\n" % (name))
+
+
+if __name__ == "__main__":
+    logging.basicConfig(
+        format="[%(process)d] %(asctime)s (%(levelname)s): %(message)s",
+        level=logging.INFO
+    )
+    if len(sys.argv) < 2 or len(sys.argv) > 3:
```
Comment (junlong4321) [inline serverless/ingest.py:183]: Can we have some instructions on the top of the script on how to operate lines 183 to 190? There seems to be some deferring sys.argv

**Comment by dextertanyj** [comment]
> LGTM. Need to check regarding 3 things.
> 
> 1. We might need to cast the test cases into a single string to render on the Submissions tab
> 2. Do I remove the questionFrontendId, titleSlug, similarQuestions, exampleTestCases
> 3. Do I commit the the Leetcode questions?

1. The casting can be done on the frontend or at the service layer in the backend but we should still keep the database normalized so that we can run the test cases more easily.
2. The question IDs and title slug can probably be removed, but we can also keep them since there's no harm to having more data scrapped.
3. I think it would be better to commit to scripts used to scrape and clean the data so that we can run the pipeline automatically in the future.

File: serverless/ingest.py (hunk 1 above)
Comment (dextertanyj) [reply serverless/ingest.py:183]: I've added a README into the `serverless` folder.

**Comment by junlong4321** [comment]
> > LGTM. Need to check regarding 3 things.
> > 
> > 1. We might need to cast the test cases into a single string to render on the Submissions tab
> > 2. Do I remove the questionFrontendId, titleSlug, similarQuestions, exampleTestCases
> > 3. Do I commit the the Leetcode questions?
> 
> 1. The casting can be done on the frontend or at the service layer in the backend but we should still keep the database normalized so that we can run the test cases more easily.
> 2. The question IDs and title slug can probably be removed, but we can also keep them since there's no harm to having more data scrapped.
> 3. I think it would be better to commit to scripts used to scrape and clean the data so that we can run the pipeline automatically in the future.

For the 3rd point I think it might be difficult to automate. I had to manually clean the output field after processing the output because the html tags between the test case output in content was inconsistent.

**Comment by dextertanyj** [comment]
> > > LGTM. Need to check regarding 3 things.
> > > 
> > > 1. We might need to cast the test cases into a single string to render on the Submissions tab
> > > 2. Do I remove the questionFrontendId, titleSlug, similarQuestions, exampleTestCases
> > > 3. Do I commit the the Leetcode questions?
> > 
> > 
> > 
> > 1. The casting can be done on the frontend or at the service layer in the backend but we should still keep the database normalized so that we can run the test cases more easily.
> > 2. The question IDs and title slug can probably be removed, but we can also keep them since there's no harm to having more data scrapped.
> > 3. I think it would be better to commit to scripts used to scrape and clean the data so that we can run the pipeline automatically in the future.
> 
> For the 3rd point I think it might be difficult to automate. I had to manually clean the output field after processing the output because the html tags between the test case output in content was inconsistent. (I split the content @ the tag "<|strong|>Output:</|strong|>", took the RHS, then split it at "\n" before taking the LHS.)
> 
> I think an alternative we can consider would be to just commit the resultant JSON file

Let's revisit this if we have time after implementing the remaining features. For now, we can distribute the JSON file through other means rather than committing data.

**Pull request merged**